web_src/css/modules/button.css

```diff
@@ -29,6 +29,12 @@
   color: inherit;
 }
 
+.ui.button.button-link:hover,
+.ui.button.button-link:focus,
+.ui.button.button-link:active {
```

Comment:
Again, this button doesn't have "underline", it doesn't navigate, so it can't be "link"

Reply:
If there is text in the button, it would have underline, and this is the intended effect if the button ever gets text inside. The browser is of course smart enough to not underline a SVG.

Checking the same button on GitHub also has `Link--onHover`, so they have a specific class for the link on hover effect, while we do it via the `button-link` class.

Reply:
It is not a link. A link has underline. You can refer to other well-designed frontend frameworks.

We already have "ghost button", I am pretty sure using "ghost button" also makes the UI look good here.

From framework's view, a usage/concept/purpose should have one clear definition/implementation.

Here is only needs a plain button with hover highlight, this usage has been covered by "ghost button".

Reply:
Even if I switch this button to `button-ghost`, this CSS needs to remain to fix the `button-link`.

Reply:
Without careful design , Gitea's framework would become more and more messy, cause more and more problems, it has happened before.

So, TBH, I do not like unclear concepts and patching the styles ad-hoc. Current framework design doesn't look good to me, but I won't block.

Reply:
To me these two button concepts are very clear and they follow [Chakra UI](https://chakra-ui.com/docs/components/button/usage#button-variants) exactly, excluding the padding, but that is something I want to tackle in the future.

Reply:
Maybe they should be documented in CSS and showed as examples in the devtest page.

Reply:
Another problem is: our "ghost button" is not the  Chakra UI's ghost button, Gitea's UI is not  Chakra UI.

Many "ghost button" in Gitea doesn't want that background or padding.

That's why I suggested not using "ghost button", but call it "plain button".

Reply:
A "plain button" alone is too unspecific. There are two competing hover/active styles, either with background fill or link underline effect. So we have two classes to let the author decide which one they want.

Reply:
To showcase, the regular render is the same:

<img width="215" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/a743fc87-38ee-41ad-8b04-c0a0614e3f78">

With hover, they differ:

<img width="212" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/db7d1e93-78c8-4adb-876b-0fc3cf4b3fe6">
<img width="214" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/87f50f4a-d6fd-4590-8d61-80c2316bb6a3">

Reply:
Making our "ghost button" has background breaks many UI.

From the beginning, these buttons are Fomantic's "tertiary" button, it doesn't have background either.

I do not see a real case why Gitea should make a button have underline as a link.

For example, why this button should have background ....

![image](https://github.com/go-gitea/gitea/assets/2114189/3c918836-7e43-4ca4-9ec4-e6ec43211fff)

And these buttons could just have hover highlight, I do not think they must have background:

![image](https://github.com/go-gitea/gitea/assets/2114189/54e64def-fd1b-4d7d-b2dd-7af161f1235f)

Reply:
First case, we need to fix by either converting to `button-link`. Second case is intentional and I think it looks great.

Reply:
If these buttons are clearly defined, have comments with these CSS class names, and have examples on the devtest UI page, and legacy problems are cleaned up, then it looks good to me.

Reply:
Yes, I'm working on another button tweak PR.

Reply:
The change here is included in https://github.com/go-gitea/gitea/pull/24989, so I'll back it out here and this PR depends on https://github.com/go-gitea/gitea/pull/24989.

Reply:
Backed out.